jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -657,6 +688,30 @@ private static void ensureProperties(boolean isInitialization) {
 	initializedProperties.put("native.encoding", platformEncoding); //$NON-NLS-1$
 	/*[ENDIF] JAVA_SPEC_VERSION >= 17 */
 
+	/*[IF JAVA_SPEC_VERSION >= 19]*/
+	/* Reinitialize required properties. */
+	if (null != setOutProp) {
+		initializedProperties.put("stdout.encoding", setOutProp);  //$NON-NLS-1$
+	} else {
+		if (null == initializedProperties.get("stdout.encoding")) {
+			String stdoutEncoding = initializedProperties.get("sun.stdout.encoding"); //$NON-NLS-1$
+			if (null != stdoutEncoding) {
+				initializedProperties.put("stdout.encoding", stdoutEncoding); //$NON-NLS-1$
+			}
+		}
+	}
+	if (null != setErrProp) {
+		initializedProperties.put("stderr.encoding", setErrProp);  //$NON-NLS-1$
+	} else {
+		if (null == initializedProperties.get("stderr.encoding")) {
+			String stderrEncoding = initializedProperties.get("sun.stderr.encoding"); //$NON-NLS-1$
+			if (null != stderrEncoding) {
+				initializedProperties.put("stderr.encoding", stderrEncoding); //$NON-NLS-1$
+			}
+		}
+	}
+	/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```

Comment:
I don't think `setOutProp` or `setErrProp` would ever be non-null here (if those fields were retained), so this can be reduced to:
```
	/*[IF JAVA_SPEC_VERSION >= 19]*/
	/* Ensure required properties are set, using legacy names. */
	String stdoutEncoding = initializedProperties.get("sun.stdout.encoding"); //$NON-NLS-1$
	if (null != stdoutEncoding) {
		initializedProperties.putIfAbsent("stdout.encoding", stdoutEncoding); //$NON-NLS-1$
	}
	String stderrEncoding = initializedProperties.get("sun.stderr.encoding"); //$NON-NLS-1$
	if (null != stderrEncoding) {
		initializedProperties.putIfAbsent("stderr.encoding", stderrEncoding); //$NON-NLS-1$
	}
	/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```

Reply:
The code handles the case where System.setProperties(null) is called by a user, which calls ensureProperties(false) after startup. In this case setOutProp or setErrProp can be non-null.